### ISSUE TITLE
Broken on windows?

### ISSUE THREAD

**Issue by clonest1ck** [kevinGodell/mse-live-player#5]
Hi, 
first of all; great work!
I am trying out your application on two different systems, macOS and Windows 10. From my mac I am able to stream the image from my display to both the Windows and mac client, but from the Windows machine I have had less luck. I have tried the ffmepg command used on its own and it spits out data just fine. My theory is that something in the forking of a new process is broken in windows maybe? I am running it in cmd, which should be no problem as both node and ffmpeg runs on their own. 

My ffmpeg command (mac):
ffmpeg -f avfoundation -probesize 32 -analyzeduration 10000 -i 1 -an -r 30 -c:v libx264 -f mp4 -g 5 -keyint_min 1 -pix_fmt yuv420p -profile:v baseline -level 3 -movflags +frag_keyframe+empty_moov+default_base_moof+omit_tfhd_offset -metadata title="ip 192.168.1.4" -reset_timestamps 1 pipe:1

Windows runs the same command, but the dshow filter is used instead and with my input set to my capture card (worked with ffmepg before). 

Besides the different ffmepg commands I can't figure out that anything else is different, even tried turning off the Windows firewall..

**Comment by kevinGodell** [comment]
Can you try to create a playable mp4 file by changing the output from pipe:1 to test.mp4 and see if that plays in VLC? We must first make sure that can happen before attempting to stream it.

**Comment by clonest1ck** [comment]
On my Windows machine I am able to create a playable mp4 file with the command
ffmpeg -f dshow -probesize 32 -analyzeduration 1000 -i video="Game Capture 4K60 Pro Video 01" -an -r 30 -c:v libx264 -f mp4 -g 10 -keyint_min 10 -pix_fmt yuv420p -profile:v baseline -level 3 -movflags +frag_keyframe+empty_moov+default_base_moof+omit_tfhd_offset -metadata title="ip 192.168.64.183" -reset_timestamps 1 test.mp4
in cmd. Using that same command in the server.js file results in no file being created.

**Comment by kevinGodell** [comment]
One thing that could be a problem is that this project is not using the latest versions of the module dependencies. All of my latest work was in my other project [ffmpeg-streamer](https://github.com/kevinGodell/ffmpeg-streamer/releases). The issue that you posted here has given me the idea to include avfoundation and dshow as input sources when I get a chance, which may be this weekend.

It would be cool if you could post a sample mp4 video online somewhere that I can download and analyze to verify that it was formatted correctly. Thanks.

**Comment by clonest1ck** [comment]
Okay, I also did some digging into the spawn() function and the child_process api in node, there seems to be some problems or requirements with it according to:
https://github.com/nodejs/node/issues/19947
but I am not convinced that this is related. 

Attached is the video sample you requested. Thank you! :)
[test.mp4.zip](https://github.com/kevinGodell/mse-live-player/files/1924121/test.mp4.zip)

**Comment by kevinGodell** [comment]
I made a quick [test](https://github.com/kevinGodell/mp4frag/blob/master/tests/test9.js) for mp4frag to verify that your mp4 was formatted correctly, and it was good. The trouble might be from `-i video="Game Capture 4K60 Pro Video 01"` with the quotes and/or spaces. I have heard from some people that sometimes nodejs chokes when spawning if the spaces in the parameters are not properly handled. Can you post a **gist** of the file that contains the code?

I have a question about your input device. I did a quick search and it seems that it is a gaming video capture card? Does it record your gameplay?

Also, I forgot to mention that the server.js code that streams my 14 cams was tested and works good on mac, linux, windows 7 and 10.  The goal was to be cross platform.

**Comment by clonest1ck** [comment]
I have discovered that the issue was, like you said, using spaces and quotes when defining my input source. Using the syntax **video="screen-capture-recorder"** I got no video but using **video=screen-capture-recorder** it worked. Now I just need to figure out how to solve this with my spaced string...

Anyhow, the banned character error seems to be a bug with the node api and not your repo. Sorry for the inconvenience! 

And once again, really nice work!
